### PR TITLE
handshake in ssl_client.cpp

### PR DESCRIPTION
**Pull request by a-c-sreedhar-reddy**
issue #2041

### CODE REVIEW

**Comment by lbernstone** [comment]
You should use a #define value rather than a hardcoded timeout value.

**Comment by atanisoft** [comment]
Instead of a #define it would be even better to have this timeout value configured in sslclient_context so it can be adjusted by the library users code.

**Comment by me-no-dev** [comment]
agreed with @atanisoft ! @a-c-sreedhar-reddy will you make the change?

**Comment by a-c-sreedhar-reddy** [comment]
> agreed with @atanisoft ! @a-c-sreedhar-reddy will you make the change?

@me-no-dev changes made.

File: libraries/WiFiClientSecure/src/WiFiClientSecure.h [hunk 1]
```diff
@@ -62,6 +62,7 @@ class WiFiClientSecure : public WiFiClient
     bool loadCertificate(Stream& stream, size_t size);
     bool loadPrivateKey(Stream& stream, size_t size);
     bool verify(const char* fingerprint, const char* domain_name);
+    void setHandshakeTimeout(unsigned long handshake_timeout);
```
Comment (atanisoft) [inline libraries/WiFiClientSecure/src/WiFiClientSecure.h]: It probably should be noted that this is expected to be seconds and not milliseconds.

Comment (a-c-sreedhar-reddy) [reply libraries/WiFiClientSecure/src/WiFiClientSecure.h]: yeah I converted them to millis in the function.

Comment (atanisoft) [reply libraries/WiFiClientSecure/src/WiFiClientSecure.h]: most likely some sort of note here will help others not to pass the wrong type. Maybe even in the method itself do some bounds checking to ensure it doesn't wait forever (ie: cap it at 3600 seconds)

Comment (a-c-sreedhar-reddy) [reply libraries/WiFiClientSecure/src/WiFiClientSecure.h]: @atanisoft  default timeout will be 2 min. It is user wish if he wants to change it.

**Pull request merged**